x-pack/plugins/reporting/public/share_context_menu/register_pdf_png_reporting.tsx

```diff
@@ -116,7 +118,7 @@ export const reportingScreenshotShareProvider = ({
     });
 
     const jobProviderOptions: JobParamsProviderOptions = {
-      shareableUrl,
+      shareableUrl: isDirty ? shareableUrl : shareableUrlForSavedObject ?? shareableUrl,
```

Comment:
I've filed this issue to remove V1 code from reporting, which will make the `shareableUrl` field not part of `JobParamsProviderOptions` in the future: https://github.com/elastic/kibana/issues/154601